src/main/java/net/minestom/server/event/player/PlayerChangeHeldSlotEvent.java

```diff
@@ -13,33 +14,77 @@
 public class PlayerChangeHeldSlotEvent implements PlayerInstanceEvent, CancellableEvent {
 
     private final Player player;
-    private byte slot;
+    private final byte oldSlot;
+    private byte newSlot;
 
     private boolean cancelled;
 
-    public PlayerChangeHeldSlotEvent(@NotNull Player player, byte slot) {
+    public PlayerChangeHeldSlotEvent(@NotNull Player player, byte oldSlot, byte newSlot) {
         this.player = player;
-        this.slot = slot;
+        this.oldSlot = oldSlot;
+        this.newSlot = newSlot;
     }
 
     /**
-     * Gets the slot which the player will held.
-     *
+     * Gets the slot number that the player is currently holding
+     * @return The slot index that the player currently is holding
+     */
+    public int getOldSlot() {
+        return oldSlot;
+    }
+
+    /**
+     * Gets the slot which the player will hold.
+     * @deprecated Use {{@link #getNewSlot()}} instead
      * @return the held slot
      */
+    @Deprecated
     public byte getSlot() {
-        return slot;
+        return newSlot;
```

Comment:
Should you really introduce a breaking change in a method that you are deprecating? Seems like a bad idea